lindi/LindiH5pyFile/LindiH5pyFile.py

```diff
@@ -14,19 +14,35 @@
 
 
 class LindiH5pyFile(h5py.File):
-    def __init__(self, _file_object: Union[h5py.File, zarr.Group]):
+    def __init__(self, _file_object: Union[h5py.File, zarr.Group], *, _zarr_store: Union[ZarrStore, None] = None, _mode: Literal["r", "r+"] = "r"):
```

Comment:
In h5py, `h5py.File` is a subclass of `h5py.Group`. Multiple inheritance is ugly but there is some code shared between `LindiH5pyFile` and `LindiH5pyGroup` that *I think* could be simplified by having `LindiH5pyFIle` inherit also from `LindiH5pyGroup`.

Reply:
I'd rather duplicate the code than to add that complexity. But it may be important to have `isinstance(lindi_file, h5py.Group)` return True. So I'll open this as a separate issue.